### PR TITLE
NetItem extensions

### PR DESCRIPTION
**Pull request by MarioE**
Implements #927. This is a breaking change, but I really doubt that anyone actually used ```NetItem``` due to its relative uselessness.

### CODE REVIEW

File: TShockAPI/TShockAPI.csproj [hunk 1]
```diff
@@ -188,7 +189,7 @@
   </PropertyGroup>
   <ProjectExtensions>
     <VisualStudio>
-      <UserProperties BuildVersion_IncrementBeforeBuild="False" BuildVersion_StartDate="2011/6/17" BuildVersion_BuildVersioningStyle="None.None.None.MonthAndDayStamp" BuildVersion_BuildAction="Both" BuildVersion_UpdateFileVersion="True" BuildVersion_UpdateAssemblyVersion="True" />
+      <UserProperties BuildVersion_UpdateAssemblyVersion="True" BuildVersion_UpdateFileVersion="True" BuildVersion_BuildAction="Both" BuildVersion_BuildVersioningStyle="None.None.None.MonthAndDayStamp" BuildVersion_StartDate="2011/6/17" BuildVersion_IncrementBeforeBuild="False" />
```
Comment (nicatronTg) [inline TShockAPI/TShockAPI.csproj]: This change is slightly unnecessary because we manually handle versioning. Can you take it out?

Comment (MarioE) [inline TShockAPI/TShockAPI.csproj]: There isn't actually a change, the stuff is just getting reordered.

Comment (nicatronTg) [inline TShockAPI/TShockAPI.csproj]: I'm blind.

**Comment by nicatronTg** [comment]
lgtm

**Comment by Enerdy** [comment]
I believe VS automatically reorders those alphabetically on build. It has happened to me, but I didn't add the .csproj file to the commit.

**Pull request merged**